File: Data Structures & Algorithms/maximum-product-subarray/submission-2.java
class Solution {
    public int maxProduct(int[] nums) {
        int[] maxProd = new int[nums.length];
        int[] minProd = new int[nums.length];
        if(nums.length == 1){
            return nums[0];
        }
        maxProd[0] = nums[0];
        minProd[0] = nums[0];
        int result = 0;
        for(int i = 1; i < nums.length; i++){
            maxProd[i] = Math.max(nums[i], Math.max(nums[i] * maxProd[i-1], nums[i] * minProd[i-1]));
            minProd[i] = Math.min(nums[i], Math.min(nums[i] * maxProd[i-1], nums[i] * minProd[i-1]));
            result = Math.max(result, maxProd[i]);
        }
        return result;
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        set_of = set(nums)

        curmax = 0

        for num in nums:
            if num - 1 not in nums:
                i = 0
                while num + i in nums:
                    i += 1
                    curmax = max(curmax, i)
                    
        return curmax


        